core/src/main/java/org/frankframework/logging/IbisMaskingLayout.java

```diff
@@ -189,44 +195,63 @@ public static Set<Pattern> getGlobalReplace() {
 		return globalReplace;
 	}
 
-	public static void cleanGlobalReplace() {
+	public static void clearGlobalReplace() {
 		globalReplace = new HashSet<>();
 	}
 
-	public static void addToThreadLocalReplace(Set<Pattern> hideRegexSet) {
+	public static void setThreadLocalReplace(@Nullable Collection<Pattern> hideRegexSet) {
 		if (hideRegexSet == null || hideRegexSet.isEmpty()) return;
 
-		if (threadLocalReplace.get() == null)
-			createThreadLocalReplace();
-
-		threadLocalReplace.get().addAll(hideRegexSet);
+		clearThreadLocalReplace();
```

Comment:
Ik denk dat je hier een potentiele NPE introduceert zonder die not null check in `clearThreadLocalReplace`

Reply:
Volgens mij niet, ik zie in ieder geval niet hoe. `clearThreadLocalReplace` hoeft die niet te hebben. In `setThreadLocalReplace` roep ik aan `getOrCreateThreadLocalReplace` en die zal nooit `null` teruggeven.